README.md

```diff
@@ -60,7 +62,106 @@ int axis = -1;
+```bash
+git clone --recursive https://github.com/CExA-project/kokkos-fft.git
+```
 
 ### CMake
-Up to now, we just rely on the CMake options for Kokkos.
+Since KokkosFFT is a header-only library, it is enough to simply add as a subdirectory. It is assumed that kokkos and kokkosFFT are placed under ```<project_directory>/tpls```.
+Here is an example to use KokkosFFT in the following CMake project. 
+```
+---/
+ |
+ └──<project_directory>/
+    |--tpls
+    |    |--kokkos/
+    |    └──kokkosFFT/
+    |--CMakeLists.txt
+    └──hello.cpp
+```
+
+The ```CMakeLists.txt``` would be 
+```CMake
+cmake_minimum_required(VERSION 3.23)
+project(kokkos-fft LANGUAGES CXX)
+
+add_subdirectory(tpls/kokkos)
+add_subdirectory(tpls/kokkos-fft)
+
+add_executable(hello-kokkos-fft hello.cpp)
+target_link_libraries(hello-kokkos-fft PUBLIC Kokkos::kokkos KokkosFFT::fft)
+```
+
+For the compilation, we basically rely on the CMake options for Kokkos. For example, the configure options for A100 GPU is as follows.
+```
+cmake -DBUILD_TESTING=ON \
+      -DCMAKE_CXX_COMPILER=<project_directory>/tpls/kokkos/bin/nvcc_wrapper \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DKokkos_ENABLE_CUDA=ON \
+      -DKokkos_ENABLE_CUDA_CONSTEXPR=ON \
+      -DKokkos_ARCH_AMPERE80=ON \
+      -DKokkos_ENABLE_CUDA_LAMBDA=On ..
+```
+This way, all the functionalities are executed on A100 GPUs.
+
+### Install as a library
+Is is assumed that the Kokkos is installed under ```<lib_dir>/kokkos``` targeting Icelake with OpenMP backend. ```Kokkos_dir``` also needs to be set appropriately. Here is a recipe to install KokkosFFT under ```<lib_dir>/kokkosFFT```.
+
+```bash
+export KOKKOSFFT_INSTALL_PREFIX=<lib_dir>/kokkosFFT
+export KokkosFFT_DIR=<lib_dir>/kokkosFFT/lib64/cmake/kokkos-fft
+
+mkdir build_KokkosFFT && cd build_KokkosFFT
+cmake -DBUILD_TESTING=OFF \
+      -DCMAKE_CXX_COMPILER=icpx \
+      -DKokkos_ENABLE_OPENMP=ON \
+      -DKokkos_ARCH_SKX=ON \
+      -DCMAKE_INSTALL_PREFIX=${KOKKOSFFT_INSTALL_PREFIX} ..
+cmake --build . -j 8
+cmake --install .
+```
+
+Here is an example to use KokkosFFT in the following CMake project. 
+```
+---/
+ |
+ └──<project_directory>/
+    |--CMakeLists.txt
+    └──hello.cpp
+```
+
+The ```CMakeLists.txt``` would be 
+```CMake
+cmake_minimum_required(VERSION 3.23)
+project(kokkos-fft LANGUAGES CXX)
+
+find_package(Kokkos CONFIG REQUIRED)
+find_package(KokkosFFT CONFIG REQUIRED)
+
+add_executable(hello-kokkos-fft hello.cpp)
+target_link_libraries(hello-kokkos-fft PUBLIC Kokkos::kokkos KokkosFFT::fft)
+```
+
+The code can be built as
+```bash
+export KOKKOSFFT_INSTALL_PREFIX=<lib_dir>/kokkosFFT
+export KokkosFFT_DIR=<lib_dir>/kokkosFFT/lib64/cmake/kokkos-fft
+
+mkdir build && cd build
+cmake -DCMAKE_CXX_COMPILER=icpx \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DKokkos_ENABLE_OPENMP=ON \
+      -DKokkos_ARCH_SKX=ON ..
+cmake --build . -j 8
```

Comment:
```suggestion
cmake -b build_dir -DCMAKE_PREFIX_PATH="<install_dir>/kokkos;<install_dir>/kokkos_fft" .
cmake --build build_dir --parallel 
```

I do not understand why `Kokkos_ENABLE_OPENMP` was passed here if Kokkos is already installed.

Reply:
That is true. May be unnecessary